### ISSUE TITLE
[BUG][BLOCKED BY UNITY SUPPORT] First PlayMode test case run fails when using `-batchmode`

### ISSUE THREAD

**Issue by ViMaSter** [Studio-Lovelies/GG-JointJustice-Unity#200]
**tl;dr:** The first (few) PlayMode test(s) will fail on GitHub until we get a response from Unity Support.

## Describe the bug

Introduced by https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/189/, the first test case that utilizes [`InputTestFixture`](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/api/UnityEngine.InputSystem.InputTestFixture.html) ([`Tests.PlayModeTests.Tools.InputTestTools`](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/274d3fce2cef8d03cac3a62ffd9339ae5e9b403a/unity-ggjj/Assets/Tests/PlayModeTests/Tools/InputTestTools.cs) in our project) fails to succeed when run headless using `-batchmode`.

## Steps To Reproduce

- Checkout 4d74b50c29d95ce96699524889358c860d912201 or any commit after it
- Run all `PlayMode` tests
- All tests should succeed
- Close all Unity editor instances
- Run `"[PATH_TO_UNITY_EDITOR]/Unity.exe" -runTests -batchmode -projectPath "[PATH_TO_THIS_REPO]/unity-ggjj" -testResults results.xml -testPlatform PlayMode` in a command line
  - for any non-Windows OS: remove the `.exe`-suffix
  - replace `[PATH_TO_UNITY_EDITOR]` and `[PATH_TO_THIS_REPO]` with both the absolute path to your Unity Editor folder and your local clone of this repository respectively
- Wait for the `Unity` process to exit 
- Check the generated `results.xml` that was generated inside your current working directory
- (At least, but possibly more than) one test should fail. This test/these tests should be (one of the) first test(s) visible inside the `Test Runner` window inside the Unity Editor

## Expected behavior

Running tests via the `Test Runner` window by hand and using `-batchmode` should yield the same test results.

## Additional context

Opened up a ticket with Unity support. Can't post the ticket link though, because that link has both the ID and some random secret in it, which gives full access to all tickets tied to my account. 👍
<img src="https://user-images.githubusercontent.com/1689033/155840520-52d8769e-6078-475f-aa96-a68f196452c9.png" width="300" />

---

From what I could gather by doing a few changes [on a branch on my fork of our repo](https://github.com/vimaster/GG-JointJustice-Unity/tree/refactor/dialogue-controller-2-fix), I seem to actually have implemented the `InputTestFixture` incorrectly, so this might be related to it:

[Based on the official (and pretty sparse) documentation](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/api/UnityEngine.InputSystem.InputTestFixture.html#:~:text=Can%20be%20derived%20from%20or%20simply%20instantiated%20from%20another%20test%20fixture.), there's two ways to use `InputTestFixture`:
- Either [having the class that contains the test cases derive from it](https://github.com/Unity-Technologies/InputSystem/blob/develop/Assets/Tests/InputSystem/Plugins/DeviceSimulatorTests.cs#L15) **OR**
- (this is what we're doing at the moment) instantiating it inside the class containing the tests

Questionable interface design aside, instantiating it in a class [requires us to also call the `Setup()` and `TearDown()`](https://forum.unity.com/threads/input-when-the-game-does-not-have-focus-for-integration-tests.991142/#post-6437402) which we currently are not doing.  
Doing this actually fixes two issues we've encountered in the past:
1. running the tests multiple times resulting in "more than 255 virtual keyboards" breaking all input and requiring Unity to be restarted
<img src="https://user-images.githubusercontent.com/1689033/155840987-69340288-dd9c-4eef-ae5f-3825ae4f8e1b.png" width="300" />

2. any sort of mouse or keyboard input disrupting test execution, requiring the Editor to have Window focus and no keyboard or mouse input to be made while tests are run

From my understanding, `Setup()` and `TearDown()` instruct Unity to either switch to or away from a context inside the `InputTestFixture` instance. Doing so, detaches the engine from any physical devices currently plugged in, so tests are actually running in isolation with only the devices present inside the `InputTestFixture`.

That said, I was unable to get mouse tests working that way. I'm sure the way I've set up these tests is probably also not the "recommended approach", resulting in this issue. More importantly, however, it also makes this bug even worse, as the Editor wouldn't just report failed tests but actually freeze and not respond generating no test results at all, as it did a few PRs ago.
